agent/lib/src/external_api/model.rs

```diff
@@ -0,0 +1,139 @@
+use serde::{Deserialize, Serialize};
+
+#[derive(Deserialize, Serialize, Debug, PartialEq, Eq)]
+pub struct FileModel {
+    pub filename: String,
+    pub content: String,
+}
+
+#[derive(Deserialize, Serialize, Debug)]
+pub struct CodeEntry {
+    pub files: Vec<FileModel>,
+    pub script: Vec<String>, // All commands to execute at startup
+}
+
+#[derive(Deserialize, Serialize, Debug, PartialEq, Eq)]
+pub enum Type {
+    Status,
+    Request,
+    Response,
+}
+
+#[derive(Deserialize, Serialize, Debug, PartialEq, Eq)]
+pub enum Code {
+    Run,
+    Ok,
```

Comment:
Is `Code` and exit code, a status code? Is `Run` meant to be `Running`? Is `Ok` meant to describe a properly terminated run? I don't know. Please add some comments to describe your API.